storybook/src/components/TextInput/TextInput.docs.mdx

```diff
@@ -14,14 +14,133 @@ import README from "../../../../packages/css/src/components/text-input/README.md
+It has built-in validation to check for a valid email format.
+However, do not rely only on the browser’s validation.
+
+The `email` input field looks like a standard text input.
+On some devices, it may show an email-specific keyboard.
+
+Consider setting the following attributes:
+
+1. Set `autocomplete="email"` to help browsers autofill the user’s email address.
+2. Set `autocapitalize="none"`, `autocorrect="off"` and `spellcheck="false"` to stop browsers automatically changing user input.
+   Email addresses shouldn’t be checked for spelling or grammar.
+   This may also prevent posing the email address to third-party plugins.
+
+Follow the [guidelines for asking for email addresses](https://design-system.service.gov.uk/patterns/email-addresses/) of the GOV.UK Design System.
+
+<Canvas of={TextInputStories.EmailAddress} />
+
+### Type: Web address
+
+This field helps the user enter a web address or URL.
+It has built-in validation to check for a valid format.
+However, do not rely only on the browser’s validation.
+
+The `url` input field looks like a standard text input.
+On some devices, it may show a URL-specific keyboard to aid in entering web addresses.
+
+Consider setting the following attributes:
+
+1. Set `autocomplete="url"` to help browsers autofill the user’s web address.
+2. Set `autocapitalize="none"`, `autocorrect="off"` and `spellcheck="false"` to stop browsers automatically changing user input.
+   Email addresses shouldn’t be checked for spelling or grammar.
+   This may also prevent posing the web address to third-party plugins.
+
+<Canvas of={TextInputStories.WebAddress} />
+
+### Type: Phone number
+
+This field helps the user enter a phone number.
+It has built-in validation to check for a valid format.
+However, do not rely only on the browser’s validation.
+
+The `tel` input field looks like a standard text input.
+On mobile devices, it may display a numeric keypad or a keyboard optimized for entering phone numbers.
+
+Guidelines:
+
+1. Set `autocomplete="tel"` to help browsers autofill the user’s phone number.
+
+Follow the [guidelines for asking for telephone numbers](https://design-system.service.gov.uk/patterns/telephone-numbers/) of the GOV.UK Design System.
+
+<Canvas of={TextInputStories.PhoneNumber} />
+
 ### Placeholder
 
+This text appears in the field when it is empty. It can give a brief hint about the kind of data to enter.
+
+Don’t try too hard to find a suitable text for a placeholder.
+Inputs without placeholder text are just fine – the label and a description should be clear enough.
+Do not use a placeholder instead of a [Label](?path=/docs/components-forms-label--docs).
+
+Follow the [guidelines for placeholder text](https://www.nldesignsystem.nl/richtlijnen/formulieren/placeholders) by NL Design System.
+
 <Canvas of={TextInputStories.Placeholder} />
 
 ### Invalid
 
+Indicates that the input value does not meet the specified constraints.
+The border of an invalid input is red.
```

Comment:
We removed this line for the Date input, do we want to remove it as well here?

Reply:
Yes, removed.